revert cfi additions to clone on i386/x86_64 to workaround problems in
gcc's unwinder code.  this is not a bug in glibc, it triggers problems
elsewhere.  this cfi code does not gain us a whole lot anyways.

http://gcc.gnu.org/ml/gcc/2006-12/msg00293.html

---
 sysdeps/unix/sysv/linux/i386/clone.S   |    4 ----
 sysdeps/unix/sysv/linux/x86_64/clone.S |    4 ----
 2 files changed, 8 deletions(-)

Index: glibc-2.8-20080811/sysdeps/unix/sysv/linux/i386/clone.S
===================================================================
--- glibc-2.8-20080811.orig/sysdeps/unix/sysv/linux/i386/clone.S
+++ glibc-2.8-20080811/sysdeps/unix/sysv/linux/i386/clone.S
@@ -120,9 +120,6 @@ L(pseudo_end):
 	ret
 
 L(thread_start):
-	cfi_startproc;
-	/* Clearing frame pointer is insufficient, use CFI.  */
-	cfi_undefined (eip);
 	/* Note: %esi is zero.  */
 	movl	%esi,%ebp	/* terminate the stack frame */
 #ifdef RESET_PID
@@ -155,7 +152,6 @@ L(nomoregetpid):
 	jmp	L(haspid)
 	.previous
 #endif
-	cfi_endproc;
 
 	cfi_startproc
 PSEUDO_END (BP_SYM (__clone))
Index: glibc-2.8-20080811/sysdeps/unix/sysv/linux/x86_64/clone.S
===================================================================
--- glibc-2.8-20080811.orig/sysdeps/unix/sysv/linux/x86_64/clone.S
+++ glibc-2.8-20080811/sysdeps/unix/sysv/linux/x86_64/clone.S
@@ -89,9 +89,6 @@ L(pseudo_end):
 	ret
 
 L(thread_start):
-	cfi_startproc;
-	/* Clearing frame pointer is insufficient, use CFI.  */
-	cfi_undefined (rip);
 	/* Clear the frame pointer.  The ABI suggests this be done, to mark
 	   the outermost frame obviously.  */
 	xorl	%ebp, %ebp
@@ -116,7 +113,6 @@ L(thread_start):
 	/* Call exit with return value from function call. */
 	movq	%rax, %rdi
 	call	HIDDEN_JUMPTARGET (_exit)
-	cfi_endproc;
 
 	cfi_startproc;
 PSEUDO_END (BP_SYM (__clone))
